tests/integration/test_issue_2415.py

```diff
@@ -0,0 +1,146 @@
+
+
+                @app.route("/")
+                def hello_world():
+                    return "Hello, World!"
+                """
+            )
+        )
+
+    pex_root = os.path.join(str(tmpdir), "pex_root")
+    pex = os.path.join(str(tmpdir), "pex")
+
+    # N.B.: Created with the following, where gevent 1.3.4 was picked as a lower bound since it
+    # 1st introduced the `from gevent import monkey; monkey.patch_all()` ssl check warning that is
+    # the subject of issue 2415:
+    #
+    # pex3 lock create \
+    #   --resolver-version pip-2020-resolver \
+    #   --pip-version latest \
+    #   --style universal \
+    #   --interpreter-constraint ">=3.8,<3.13" \
+    #   --indent 2 \
+    #   flask \
+    #   "gevent>=1.3.4" \
+    #   gunicorn
+    lock = data.path("locks", "issue-2415.lock.json")
+
+    run_pex_command(
+        args=[
+            "--pex-root",
+            pex_root,
+            "--runtime-pex-root",
+            pex_root,
+            "--lock",
+            lock,
+            "-M",
+            "app",
+            "-c",
+            "gunicorn",
+            "--inject-args",
+            "app:app",
+            "-o",
+            pex,
+        ],
+        cwd=str(tmpdir),
+    ).assert_success()
+
+    log = os.path.join(str(tmpdir), "log")
+    os.mkfifo(log)
+
+    @attr.s
+    class LogScanner(object):
+        port_seen = attr.ib(factory=Event, init=False)  # type: Event
+        _port = attr.ib(default=None)  # type: Optional[int]
+
+        def scan_log(self):
+            # type: () -> None
+
+            with open(log) as log_fp:
+                for line in log_fp:
+                    if self._port is None:
+                        match = re.search(r"Listening at: http://127.0.0.1:(?P<port>\d{1,5})", line)
+                        if match:
+                            self._port = int(match.group("port"))
+                            self.port_seen.set()
+
+        @property
+        def port(self):
+            # type: () -> int
+            self.port_seen.wait()
+            assert self._port is not None
+            return self._port
+
+    log_scanner = LogScanner()
+    log_scan_thread = threading.Thread(target=log_scanner.scan_log)
+    log_scan_thread.daemon = True
+    log_scan_thread.start()
+
+    with open(os.path.join(str(tmpdir), "stderr"), "wb+") as stderr_fp:
+        gunicorn = subprocess.Popen(
+            args=[pex, "--bind", "127.0.0.1:0", "--log-file", log], stderr=stderr_fp
+        )
+        atexit.register(gunicorn.kill)
+
+        with URLFetcher().get_body_stream(
+            "http://127.0.0.1:{port}".format(port=log_scanner.port)
+        ) as http_fp:
+            assert b"Hello, World!" == http_fp.read().strip()
+
+        gunicorn.kill()
+        log_scan_thread.join()
+        stderr_fp.flush()
+        stderr_fp.seek(0)
+        stderr = stderr_fp.read()
+        assert b"MonkeyPatchWarning: Monkey-patching ssl after ssl " not in stderr, stderr.decode(
```

Comment:
The failure at this commit looks like:
```
>           assert b"MonkeyPatchWarning: Monkey-patching ssl after ssl " not in stderr, stderr.decode(
                "utf-8"
            )
E           AssertionError: /tmp/pytest-of-jsirois/pytest-52/test_gevent_monkeypatch0/pex_root/unzipped_pexes/d0b0f145223b1d7c391f024af1f2baa4bebd475f/app.py:2: MonkeyPatchWarning: Monkey-patching ssl after ssl has already been imported may lead to errors, including RecursionError on Python 3.6. It may also silently lead to incorrect behaviour on Python 3.7. Please monkey-patch earlier. See https://github.com/gevent/gevent/issues/1016. Modules that had direct imports (NOT patched): ['pex.fetcher (/tmp/pytest-of-jsirois/pytest-52/test_gevent_monkeypatch0/pex_root/unzipped_pexes/d0b0f145223b1d7c391f024af1f2baa4bebd475f/.bootstrap/pex/fetcher.py)'].
E               monkey.patch_all()
```